pkg/backup/backup_compaction.go

```diff
@@ -175,7 +194,7 @@ func compactIncrementals(
 		}),
 	)
 	backupManifest, newDetails, err := prepareCompactedBackupMeta(
```

Comment:
what's the motivation behind managing a separate compaction_details field in the backup details proto, if we're going to continue to use the backup_details proto? Would it be simpler to continue using the backup_details proto, and to simply add a `compaction` bool to the the backup_details proto?

If you're concerned that certain fields in backup_details proto should be left empty during a compaction job, could we add a test to assert that those fields remain empty?

I also don't think it would be wise to copy over all fields from backup_details into compaction_details that are required, as we'll need to make sure we always update both.

It may be easier to discuss this over a call.

Reply:
I did also consider having a `compaction` bool, I'm not strongly for either or. For me, since they feel like pretty distinct jobs, I felt that having a separate struct within would be a bit clearer for the developer. Can see more on this discussion in [this Slack thread](https://cockroachlabs.slack.com/archives/C03JCUUSCD6/p1738967909217809?thread_ts=1738967369.023289&cid=C03JCUUSCD6).